sdk/appconfiguration/app-configuration/samples/react-sample/src/components/Page.tsx

```diff
@@ -0,0 +1,200 @@
+/* eslint-disable jsx-a11y/anchor-is-valid */
+/*
+  Copyright (c) Microsoft Corporation.
+  Licensed under the MIT license.
+*/
+
+import React from "react";
+import { useEffect, useState } from "react";
+import { getEnvironmentVariable } from "../utils";
+import {
+  AppConfigurationClient,
+  featureFlagPrefix,
+  isFeatureFlag,
+  isFeatureFlagClientFilter
+} from "@azure/app-configuration";
+import { InteractiveBrowserCredential } from "@azure/identity";
+
+let client: AppConfigurationClient;
+
+const featureFlag1 = {
+  id: "react-app-feature-1",
+  description: "",
+  enabled: false,
+  conditions: {
+    client_filters: []
+  }
+};
+const featureFlag2 = {
+  id: "react-app-feature-2",
+  description: "",
+  enabled: true,
+  conditions: {
+    client_filters: [
+      {
+        name: "Microsoft.TimeWindow",
+        parameters: {
+          Start: "Mon, 12 Apr 2021 07:00:00 GMT",
+          End: "Wed, 14 Apr 2021 07:00:00 GMT"
+        }
+      }
+    ]
+  }
+};
+
+export default function Page(): JSX.Element {
+  const feature1Name = "react-app-feature-1";
+  const feature2Name = "react-app-feature-2";
+  const [feature1, setFeature1] = useState<{ enabled: boolean }>({ enabled: false });
+  const [feature2, setFeature2] = useState<{ enabled: boolean }>({ enabled: false });
+
+  const getFeatureFlags = async (keys: string[]): Promise<void> => {
+    const [setting1, setting2] = await Promise.all(
+      keys.map((key: string) =>
+        client.getConfigurationSetting({
+          key: featureFlagPrefix + key
+        })
+      )
+    );
+    if (isFeatureFlag(setting1)) {
+      // console.log(`${setting1.key} is enabled : ${setting1.enabled}`, setting1);
+      setFeature1({ enabled: setting1.enabled });
+    }
+    if (isFeatureFlag(setting2)) {
+      // console.log(`${setting2.key} is enabled : ${setting2.enabled}`, setting2);
+      const clientFilter = setting2.conditions.clientFilters?.[0];
+      if (isFeatureFlagClientFilter("timeWindow", clientFilter)) {
+        const now = Date.now();
+        const withinRange =
+          now - Date.parse(clientFilter.parameters.start) > 0 &&
+          Date.parse(clientFilter.parameters.end) - now > 0;
+        setFeature2({ enabled: withinRange });
+      }
+    }
+  };
+
+  useEffect(() => {
+    const endpoint = getEnvironmentVariable("REACT_APP_APPCONFIG_ENDPOINT");
+    const clientId = getEnvironmentVariable("REACT_APP_AZURE_CLIENT_ID");
+    const tenantId = getEnvironmentVariable("REACT_APP_AZURE_TENANT_ID");
+    const credential = new InteractiveBrowserCredential({ clientId, tenantId });
```

Comment:
Using IBC is a bit worrying to me. Newing up IBC in an effect hook will work fine for a sample using the popup-based flow, but it doesn't model what we would want a user to actually do. Could we use `msal-react` and wrap its access token mechanism in a `TokenCredential` instead? Long term I think this is what we'll want to do for React users since the MSAL folks have been working on a good story here.

Reply:
Trying out msal-react to see how it fits into our story is something we definitely want to do. Might as well do it here! But if we have come close to wrapping up this PR, we can try out msal-react in a separate PR

Reply:
https://github.com/Azure/azure-sdk-for-js/issues/14988 logged.
I guess this is a broader conversation and not just limited to app-config since we have react samples for other services too.